cloudkeeper-core/cloudkeeper-dsl/src/main/java/com/svbio/cloudkeeper/dsl/DSLPluginDescriptor.java

```diff
@@ -15,7 +15,11 @@
  * <p>An instance of this class provides the {@link Class} instances (as specified by the DSL) that make up a
  * CloudKeeper plug-in declaration.
  */
-public class DSLPluginDescriptor {
+public final class DSLPluginDescriptor {
+    private DSLPluginDescriptor() {
+        throw new AssertionError(String.format("No %s instances for you!", getClass().getName()));
+    }
+
```

Comment:
You are right, but checkstyle is not smart enough to realize that there is a private constructor and therefore effectively it is a final class. Thus unless "final" is added checkstyle complains with the following error:
```
[ERROR] src/main/java/com/svbio/cloudkeeper/dsl/DSLPluginDescriptor.java:[18] (design) FinalClass: Class DSLPluginDescriptor should be declared as final.
```